File: typescript/questions/maxSubArray.ts
function maxSubArray(nums: number[]): number {
  let max = nums[0];
  let cur = nums[0];
  for (let i=1; i<nums.length; ++i) {
    cur = Math.max(cur+nums[i], nums[i]);
    max = Math.max(cur, max);
  }
  return max;
};

File: rust/questions/max_sub_array.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut max = nums[0];
        let mut cur = nums[0];
        for i in 1..nums.len() {
            cur = nums[i].max(cur+nums[i]);
            max = max.max(cur);
        }
        max
    }
}
